dot/core/service.go

```diff
@@ -635,3 +666,20 @@ func (s *Service) GetReadProofAt(block common.Hash, keys [][]byte) (
 
 	return block, proofForKeys, nil
 }
+
+func (s *Service) buildTransaction(rt runtime.Instance, ext types.Extrinsic) (types.Extrinsic, error) {
+	runtimeVersion, err := rt.Version()
+	if err != nil {
+		return types.Extrinsic{}, err
+	}
+
+	txQueueVersion := runtimeVersion.TaggedTransactionQueueVersion(runtimeVersion)
+	var externalExt types.Extrinsic
+	if txQueueVersion == 3 {
```

Comment:
would be nice to have the `3` as a named const
```suggestion
        const supportedTxVersion = 3
	if txQueueVersion ==  supportedTxVersion {
```

Reply:
See my previous comment https://github.com/ChainSafe/gossamer/pull/2562#discussion_r919253153

We should define the constants at global scope so someone doesn't re-inline them in another function in the future, especially since these could be used in different places.

> I wonder if there is a difference here when the const is defined inline.

Performance wise, no difference, so it's nice for only-local-constant-not-shared-anywhere-else